### PR TITLE
Fix error build

### LINKED ISSUE #2: Fix Build Error
- Fix some errors when build AdminGui
- Try to make BaseCode as an package of npm. Then we can remove source code of BaseCode out of solution

### LINKED ISSUE #7: Download RhetosBuild by build script
- Add download RhetosBuild in batch files
- Remove some unnecessary file in RhetosBuild (like RhetosPackages.config)
- Build Project with latest version and check errors.

### PR DESCRIPTION
**Pull request by huynguyen-95**
This fixes #2 and fixes #7 .

- Remove Basecode from solution
- Add Basecode as npm package
- Fix Build Error

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -1,6 +1,3 @@
-[submodule "2CS.BaseCode"]
-	path = 2CS.BaseCode
-	url = https://2cs.visualstudio.com/2CS.BaseCode/_git/2CS.BaseCode
 [submodule "2CS.RhetosBuild"]
```
Comment (ngoquoc) [inline .gitmodules]: 2CS.RhetosBuild submodule should be removed as well. The idea is it does not depend on any submodule from internal source control systems.

File: AdminGUI/Rhetos.AdminGuiCompile.nuspec [hunk 2]
```diff
@@ -13,14 +13,8 @@
...
-	<file src="\node_modules\@angular\**\*.d.ts" target="Resources\node_modules\@angular" />
-	<file src="\node_modules\moment\**\*.d.ts" target="Resources\node_modules\moment" />
-	<file src="\node_modules\ng2-file-upload\**\*.d.ts" target="Resources\node_modules\ng2-file-upload" />
-	<file src="\node_modules\primeng\**\*.d.ts" target="Resources\node_modules\primeng" />
-	<file src="\node_modules\reflect-metadata\**\*.d.ts" target="Resources\node_modules\reflect-metadata" />
-	<file src="\node_modules\rxjs\**\*.d.ts" target="Resources\node_modules\rxjs" />
-	<file src="\node_modules\zone.js\**\*.d.ts" target="Resources\node_modules\zone.js" />
-	<file src="\typings\**\*" target="Resources\typings" />
+	<file src="\wwwroot\lib\**\*" target="Resources\node_modules" />
+	<!-- <file src="\typings\**\*" target="Resources\typings" /> -->
```
Comment (ngoquoc) [inline AdminGUI/Rhetos.AdminGuiCompile.nuspec]: can this line be removed?

File: AdminGUI/external_scripts/tag-autocomplete-jquery.control.js [hunk 3]
```diff
@@ -0,0 +1,162 @@
+$.widget("ui.autocomplete", $.ui.autocomplete, {
```
Comment (ngoquoc) [inline AdminGUI/external_scripts/tag-autocomplete-jquery.control.js]: is there any min version of this lib?

File: AdminGUI/gulpfile.js [hunk 4]
```diff
@@ -26,36 +56,47 @@ var paths = {
...
+    { basePath: paths.npm, relPath: "@angular/router/bundles/router.umd.min.js" },
+    { basePath: paths.npm, relPath: "@angular/forms/bundles/forms.umd.min.js" },
+	
+	{ basePath: paths.npm, relPath: "basecode/bundles/basecode.min.js" },
+	// { basePath: paths.npm, relPath: "basecode/bundles/basecode.umd.min.js" },
+	
     { basePath: paths.npm, relPath: "es6-shim/es6-shim.min.js" },
     { basePath: paths.npm, relPath: "zone.js/dist/zone.js" },
-    { basePath: paths.npm, relPath: "rxjs/bundles/Rx.min.js" },
+	// { basePath: paths.npm, relPath: "rxjs/bundles/Rx.min.js" },
```
Comment (ngoquoc) [inline AdminGUI/gulpfile.js]: commented out lines should be removed.

File: AdminGUI/gulpfile.js [hunk 5]
```diff
@@ -88,58 +129,53 @@ gulp.task('compile-ts', function() {
...
+	gulp.src("./scripts/systemjs.config.js").pipe(gulp.dest('wwwroot/js/'));
+    //gulp.src("./node_modules/.tmp/Rx.umd.min.js").pipe(gulp.dest('wwwroot/lib/rxjs'));
+	
 });
 
 gulp.task("appCopy", function () {
-    gulp.src("dist/*").pipe(gulp.dest(paths.appjs));
-    gulp.src("./scripts/systemjs.config.js").pipe(gulp.dest(paths.appjs));
-    gulp.src("Scripts/**/*").pipe(gulp.dest("wwwroot/Scripts/"));
+	//Comment by Duong 17/10
```
Comment (ngoquoc) [inline AdminGUI/gulpfile.js]: I don't think such type of comment help :D

File: AdminGUI/package.json [hunk 6]
```diff
@@ -2,39 +2,52 @@
...
     "gulp-typescript": "",
-    "rimraf": "2.2.8"
+    "gulp-util": "",
+    "mkdirp": "",
+    "pure-cjs": "",
+    "rimraf": "2.2.8",
+    "source-map": "",
+    "systemjs-builder": "",
+    "traceur": "",
+    "typescript": "^2.0.3",
```
Comment (ngoquoc) [inline AdminGUI/package.json]: typescript version is not consistent with the one specified in `dependencies` section. Will that be an issue?

File: AdminGUI/scripts/controls/header-admingui.control.ts [hunk 7]
```diff
@@ -45,13 +43,13 @@ export class HeaderAdminGuiAppComponent {
      * Log Out
      * @param 
      */
-    logOut() {
-        var isLogOut = confirm("Are you sure you want to logout?");
-        if (isLogOut) {
-            this.loginService.logout();
-            window.location.href = location.origin + (location.pathname.split('/').splice(0, location.pathname.split('/').length - 3).join('/')) + '/Resources/AspNetFormsAuth/Login.html?ReturnUrl=%2f';
+    //logOut() {
```
Comment (ngoquoc) [inline AdminGUI/scripts/controls/header-admingui.control.ts]: Commented out code is kind of scary to readers (at least to me). Can it be completely removed?

**Review by ngoquoc — CHANGES_REQUESTED** [review]
Changes look ok to me. The biggest issue I can see is there are so many commented out code blocks. I think it should be either removed OR left with meaningful descriptions why it is commented out.

File: .gitmodules (hunk 1 above)
Comment (huynguyen-95) [reply .gitmodules]: Actually, we wont have submodules anymore. I will delete git submodules from repo.

File: AdminGUI/package.json (hunk 6 above)
Comment (huynguyen-95) [reply AdminGUI/package.json]: It should be fixed to 2.0.3.

File: AdminGUI/external_scripts/tag-autocomplete-jquery.control.js (hunk 3 above)
Comment (huynguyen-95) [reply AdminGUI/external_scripts/tag-autocomplete-jquery.control.js]: I checked it, and we dont use this lib anymore in AdminGui. This lib is only use int BaseCode.

File: AdminGUI/scripts/controls/header-admingui.control.ts (hunk 7 above)
Comment (huynguyen-95) [reply AdminGUI/scripts/controls/header-admingui.control.ts]: done

**Review by ngoquoc — APPROVED** [review]
LGTM

**Pull request merged**